atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueService.java

```diff
@@ -260,25 +263,27 @@ public Void apply(DSLContext ctx) {
         if (Iterables.isEmpty(rows)) {
             return ImmutableMap.of();
         }
-        return run(new Function<DSLContext, Map<Cell, Value>>() {
-            @Override
-            public Map<Cell, Value> apply(DSLContext ctx) {
-                Select<? extends Record> query = getLatestTimestampQuerySomeColumns(
-                        ctx,
-                        tableRef,
-                        ImmutableList.copyOf(rows),
-                        columnSelection.getSelectedColumns(),
-                        timestamp);
-                Result<? extends Record> records = fetchValues(ctx, tableRef, query);
-                Map<Cell, Value> results = Maps.newHashMapWithExpectedSize(records.size());
-                for (Record record : records) {
-                    results.put(
-                            Cell.create(record.getValue(A_ROW_NAME), record.getValue(A_COL_NAME)),
-                            Value.create(record.getValue(A_VALUE), record.getValue(A_TIMESTAMP)));
-                }
-                return results;
-            }
-        });
+
+        Map<Cell, Value> toReturn = new HashMap<>();
+        for (List<byte[]> part : Iterables.partition(rows, PARTITION_SIZE)) {
```

Comment:
As above, this one is called through `getRowsPartition()` which is called through `getRows()`, so the list should already have at most `PARTITION_SIZE` elements. So I don't think we need this outer loop, or this was supposed to read `batchSizeForReads`.